ring/batch.go

```diff
@@ -55,8 +79,11 @@ func (i *itemTracker) recordError(err error) int32 {
 //
 // cleanup() is always called, either on an error before starting the batches or after they all finish.
 //
+// optionalErrorFilters represent a set of conditions that an error returned
+// by the callback function should satisfy in order to be treated as a client error.
```

Comment:
This sounds like error needs to satisfy all of them. I'd suggest to support single function only, and call it `isClientError`.

I'd suggest to create new `DoBatchWithClientError` function, and implement `DoBatch` by calling `DoBatchWithClientError` with `isHTTPStatus4xx` function.

Reply:
I have applied the required changes.